punchcut/src/lib.rs

```diff
@@ -0,0 +1,200 @@
+/*!
+Glyph loading.
+*/
+
+#![forbid(unsafe_code)]
+// TODO: this is temporary-- remove when hinting is added.
+#![allow(dead_code, unused_imports, unused_variables)]
+
+/// Re-export of peniko crate.
+pub use peniko;
+
+mod error;
+mod outline;
+mod scaler;
+
+pub mod source;
+
+/// Representations of fonts and font collections.
+pub mod font {
+    pub use read_fonts::{types::Tag, CollectionRef, FileRef, FontRef, TableProvider};
+}
+
+use font::Tag;
+use source::*;
+
+use core::str::FromStr;
+
+pub use error::*;
+pub use outline::*;
+pub use scaler::*;
+
+/// Modes for hinting.
+///
+/// Only the `glyf` source supports all hinting modes.
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+#[allow(dead_code)]
+pub enum Hinting {
+    /// "Full" hinting mode. May generate rough outlines and poor horizontal
+    /// spacing.
+    Legacy,
```

Comment:
Agree with @cmyr's related feedback on the naming in the C-API draft. If these are taken from FreeType - I'd be very happy if we could make at least "Legacy" and "Modern" more descriptive.

Reply:
I made suggestions on Colin's comment above. I wonder if those feel better to you?